### PR TITLE
Fraction number of packs fe

### LINKED ISSUE #643: Fractional number of packs FE changes
Part of #505 

- Change number of packs input to allow decimals

### PR DESCRIPTION
**Pull request by roxy-dao**
Closes #643

- Showed fractional number of packs as decimals. 
- Allow editable decimal number of packs for inbound shipments and stocktake. 
- Kept outbound shipment number of packs whole number - [discussion for this](https://github.com/openmsupply/open-msupply/issues/648)

### CODE REVIEW

File: client/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonNegativeDecimalCell.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const updater = useDebounceCallback(column.setter, [column.setter], 250);
+
+  return (
+    <NonNegativeNumberInput
+      disabled={isDisabled}
+      InputProps={{ sx: { '& .MuiInput-input': { textAlign: 'right' } } }}
+      type="number"
+      value={buffer}
+      onChange={newValue => {
+        const intValue = Math.round(newValue * 100) / 100;
```
Comment (mark-prins) [inline client/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonNegativeDecimalCell.tsx:27]: :nit it isn't an integer value now.. so the variable name of `intValue` could be misleading for people reading later who don't look at the functionality carefully

File: client/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonNegativeDecimalCell.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { CellProps } from '../../../columns';
+import { NonNegativeNumberInput } from '@common/components';
+import { RecordWithId } from '@common/types';
+import { useBufferState, useDebounceCallback } from '@common/hooks';
+
+// where NonNegative is n >=0
+export const NonNegativeDecimalCell = <T extends RecordWithId>({
```
Comment (mark-prins) [inline client/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonNegativeDecimalCell.tsx:8]: concerned about the duplication between this component and the `NonNegativeIntegerCell` but it is unnecessarily complicated to add in a prop for the value rounding. best to keep what you have I think

**Review by mark-prins — APPROVED** [review]
👍  very nice! thanks - that's great.
I was thinking about limiting users - to only allow input of whole number pack sizes, and then send as float to the API - now I'm thinking not to do that. You'd have to attempt a decimal to know that it's possible.. so most likely users will just enter integers, since that's easier. and this allows decimals if required

File: client/packages/common/src/intl/number/IntlNumber.ts [hunk 3]
```diff
@@ -12,5 +12,11 @@ export const useFormatNumber = () => {
       });
       return intl.format(value ?? 0);
     },
+    twoDp: (value?: number, dp?: number): string => {
```
[the remark this answers is not in the document]
Comment (roxy-dao) [reply client/packages/common/src/intl/number/IntlNumber.ts]: [5ddcf58](https://github.com/openmsupply/open-msupply/pull/647/commits/5ddcf5862cbb4074680428269c6a8e8dc9971f90)

**Pull request merged**